plugins/fetcher/prometheus/fetcher.go

```diff
@@ -0,0 +1,146 @@
+	StaticConfigs  []map[string]interface{} `yaml:"static_configs" mapstructure:"static_configs"`
+	MetricsPath    string                   `yaml:"metrics_path,omitempty" mapstructure:"metrics_path,omitempty"`
+}
+
+// Fetcher is the struct for Prometheus fetcher
+type Fetcher struct {
+	config.CommonFields
+	// config is the top level configuratScrapeConfigsMapion of prometheus
+	ScrapeConfigsMap []*scrapeConfig `mapstructure:"scrape_configs" yaml:"scrape_configs"`
+
+	ScrapeConfigs []*promConfig.ScrapeConfig
+	// events
+	OutputEvents event.BatchEvents
+	// outputChannel
+	OutputChannel chan *v1.SniffData
+
+	cancelFunc context.CancelFunc
+}
+
+func (f *Fetcher) Name() string {
+	return Name
+}
+
+func (f *Fetcher) Description() string {
+	return "This is a fetcher for Skywalking prometheus metrics format, " +
+		"which will translate Prometheus metrics to Skywalking meter system."
+}
+
+func (f *Fetcher) DefaultConfig() string {
+	return `
+## some config here
+scrape_configs:
+ - job_name: 'prometheus'
+   metrics_path: '/metrics'
+   scrape_interval: 10s
+   static_configs:
+   - targets: ['127.0.0.1:2020']
+`
+}
+
+func (f *Fetcher) Prepare() {}
+
+func (f *Fetcher) Fetch() event.BatchEvents {
+	ctx, cancel := context.WithCancel(context.Background())
+	f.cancelFunc = cancel
+	// yaml
+	configDeclare := make(map[string]interface{})
+	configDeclare["scrape_configs"] = f.ScrapeConfigsMap
+	configBytes, err := yaml.Marshal(configDeclare)
+	if err != nil {
+		log.Logger.Fatal("prometheus fetcher configure failed", err.Error())
+	}
+	log.Logger.Debug(string(configBytes))
+	configStruct, err := promConfig.Load(string(configBytes))
+	if err != nil {
+		log.Logger.Fatal("prometheus fetcher configure load failed", err.Error())
+	}
+	f.ScrapeConfigs = configStruct.ScrapeConfigs
+	return fetch(ctx, f.ScrapeConfigs, f.OutputChannel)
+}
+
+func (f *Fetcher) Channel() <-chan *v1.SniffData {
+	return f.OutputChannel
+}
+
+func (f *Fetcher) Shutdown(context.Context) error {
+	f.cancelFunc()
+	return nil
+}
+
+func fetch(ctx context.Context, scrapeConfigs []*promConfig.ScrapeConfig, outputChannel chan *v1.SniffData) event.BatchEvents {
+	// config of scraper
+	c := make(map[string]discovery.Configs)
+	for _, v := range scrapeConfigs {
+		c[v.JobName] = v.ServiceDiscoveryConfigs
+	}
+	// manager
+	manager := discovery.NewManager(ctx, nil)
+	if err := manager.ApplyConfig(c); err != nil {
+		log.Logger.Errorf("prometheus discovery config error %s", err.Error())
+	}
+	// manager start
+	go func() {
+		if err := manager.Run(); err != nil {
+			log.Logger.Errorf("Discovery manager run failed, error %s", err.Error())
+		}
+	}()
+	// queue store
+	qs := NewQueueStore(ctx, true, Name, outputChannel)
+	scrapeManager := scrape.NewManager(nil, qs)
+	qs.SetScrapeManager(scrapeManager)
+	cfg := &promConfig.Config{ScrapeConfigs: scrapeConfigs}
+	if err := scrapeManager.ApplyConfig(cfg); err != nil {
+		log.Logger.Errorf("scrape failed, error: %s", err.Error())
```

Comment:
why we don't need to return when error？

Reply:
You are right, need to return here. but I think use log.fatal is better.